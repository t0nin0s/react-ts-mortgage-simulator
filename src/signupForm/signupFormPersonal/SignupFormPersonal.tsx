import * as React from 'react';
import {
  FormGroup,
  FormControl
} from 'react-bootstrap';
import Card from '../../components/Card/Card';
import * as FontAwesome from 'react-fontawesome';
const styles = require('../signupForm.css');

interface SignupFormPersonalProps {
    title: string;
    name: string;
    surname: string;
    handleOnChange: (event: any) => void;
    handleNext: () => void;
}

class SignupFormPersonal extends React.Component<SignupFormPersonalProps, { error: boolean } > {
  constructor() {
    super();
    this.state = {
      error: false
    };
  }

  handleSubmit = (event: Event) => {
    event.preventDefault();
  }

  validate = () => {
    let error = false;
    let inputs = {
      title: this.props.title,
      name: this.props.name,
      surname: this.props.surname
    };
    Object.entries(inputs).forEach(([key, value]) => {
      if (!value) {
        error = true;
        this.setState({
          error: error
        });
      }
    });

    if (!error) {
      this.props.handleNext();
    }
  }

  render () {
    return (
      <Card
        formTitle="Sign up"
        icon="address-card-o"
        footer={true}
      >
        <div className={styles.form__question}>
          Please tell us your name...
        </div>
        <form className={styles.form__body} onSubmit={() => this.handleSubmit}>
          <FormGroup>
            <FormControl
              value={this.props.title}
              onChange={this.props.handleOnChange}
              id="title"
              type="String"
              componentClass="select"
              placeholder="Select your title"
            >
              <option value="">Select your title</option>
              <option value="Mr">Mr</option>
              <option value="Mrs">Mrs</option>
            </FormControl>
          </FormGroup>
          <FormGroup>
            <FormControl
              value={this.props.name}
              onChange={this.props.handleOnChange}
              className="form-control"
              id="name"
              type="string"
              placeholder="First Name"
            />
          </FormGroup>
          <FormGroup>
            <FormControl
              value={this.props.surname}
              onChange={this.props.handleOnChange}
              className="form-control"
              id="surname"
              type="string"
              placeholder="Surname"
            />
          </FormGroup>
          { this.state.error &&
            <div className={styles.error_message}>
              Please select first your gender...
            </div>
          }
          <div className={styles.form__submit}>
            <FontAwesome
              name="arrow-circle-right"
              size="3x"
              style={{ color: '#ECECEC'}}
              onClick={this.validate}
            />
          </div>
        </form>
      </Card>
    );
  }
}

export default SignupFormPersonal;
