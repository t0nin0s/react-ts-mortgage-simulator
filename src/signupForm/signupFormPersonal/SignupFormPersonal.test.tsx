import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SignupFormPersonal from './SignupFormPersonal'

describe('SignupFormPersonal renders as expected', () => {
  it('should render as expected', () => {
    const component = shallow(
      <SignupFormPersonal
        title=''
        name=''
        surname=''
        handleOnChange= {jest.fn()}
        handleNext= {jest.fn()} />
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
