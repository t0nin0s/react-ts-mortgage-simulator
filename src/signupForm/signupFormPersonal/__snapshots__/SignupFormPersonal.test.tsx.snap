// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupFormPersonal renders as expected should render as expected 1`] = `
<Card
  footer={true}
  formTitle="Sign up"
  icon="address-card-o"
>
  <div>
    Please tell us your name...
  </div>
  <form
    onSubmit={[Function]}
  >
    <FormGroup
      bsClass="form-group"
    >
      <FormControl
        bsClass="form-control"
        componentClass="select"
        id="title"
        onChange={[Function]}
        placeholder="Select your title"
        type="String"
        value=""
      >
        <option
          value=""
        >
          Select your title
        </option>
        <option
          value="Mr"
        >
          Mr
        </option>
        <option
          value="Mrs"
        >
          Mrs
        </option>
      </FormControl>
    </FormGroup>
    <FormGroup
      bsClass="form-group"
    >
      <FormControl
        bsClass="form-control"
        className="form-control"
        componentClass="input"
        id="name"
        onChange={[Function]}
        placeholder="First Name"
        type="string"
        value=""
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
    >
      <FormControl
        bsClass="form-control"
        className="form-control"
        componentClass="input"
        id="surname"
        onChange={[Function]}
        placeholder="Surname"
        type="string"
        value=""
      />
    </FormGroup>
    <div>
      <FontAwesome
        name="arrow-circle-right"
        onClick={[Function]}
        size="3x"
        style={
          Object {
            "color": "#ECECEC",
          }
        }
      />
    </div>
  </form>
</Card>
`;
